core/src/main/java/io/kestra/core/models/executions/TaskRun.java

```diff
@@ -220,4 +217,25 @@ public String toStringState() {
             ", state=" + this.getState().getCurrent().toString() +
             ")";
     }
+
+    /**
+     *
+     * @param task Contains the retry configuration
+     * @return The next retry date, null if maxAttempt is reached
+     */
+    public Instant nextRetryDate(Task task) {
```

Comment:
There is two concepts here:
- Whether we should retry
- When to retry.

For better code clarity and comprehension I would have separate the two: `shouldBeRetried()` and `nextRetryDate()`.
Looking at the code, most of the time you want to know if the task run should be retried so it makes sense to split the method in two.